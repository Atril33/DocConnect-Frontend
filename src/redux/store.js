import { configureStore } from '@reduxjs/toolkit';
import authReducer from './auth/authSlice';
import doctorReducer from './doctors/doctorSlice';

export const store = configureStore({
  reducer: {
    auth: authReducer,
    doctors: doctorReducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: false,
  }),
});

const selectAuth = (store) => store.auth;

export {
  selectAuth,
};

export default store;
