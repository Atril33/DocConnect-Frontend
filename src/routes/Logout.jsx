import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link, useNavigate } from 'react-router-dom';
import { toast } from 'react-toastify';
import { useForm } from 'react-hook-form';
import { selectAuth } from '../redux/store';
import { logoutUser } from '../redux/auth/authActions';
import 'react-toastify/dist/ReactToastify.css';
import Button from '../components/Button';

const Logout = () => {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const { handleSubmit } = useForm();
  const { error } = useSelector(selectAuth);

  const handleLogOut = () => {
    toast.promise(
      dispatch(logoutUser()),
      {
        pending: 'loading...',
        error,
        success: 'Logout!',
      },
    ).then(() => navigate('/'));
  };

  if (error) {
    toast.error(`Oops, something went wrong: ${error}`);
  }

  return (
    <div
      className="relative flex items-center justify-center h-screen bg-center bg-cover bg-home-background-mobile md:bg-home-background"
    >
      <div className="w-full max-w-xs">
        <h1 className="mb-6 text-5xl font-bold text-center text-white">Logout</h1>
        <form className="px-8 pt-6 pb-8 mb-4 bg-white rounded shadow-md" onSubmit={handleSubmit(handleLogOut)}>
          <div className="mb-4">
            <p>Are you sure you want to log out?</p>
          </div>
          <div className="mb-4">
            <Button className="w-full" text="Log out" type="submit" />
          </div>
          <p className="text-xs text-center text-gray-500">OR</p>
          <p className="text-xs text-center text-gray-500">
            Go back
            {' '}
            <Link to="/" className="underline">Home</Link>
          </p>
        </form>
      </div>
    </div>
  );
};
export default Logout;
