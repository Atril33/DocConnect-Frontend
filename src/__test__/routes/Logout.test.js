import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { QueryClient, QueryClientProvider } from 'react-query';
import { Provider } from 'react-redux';
import { store } from '../../redux/store';
import Logout from '../../routes/Logout';

test('Logout component renders correctly', () => {
  const queryClient = new QueryClient();
  const component = renderer.create(
    <Router>
      <QueryClientProvider client={queryClient}>
        <Provider store={store}>
          <Logout />
        </Provider>
      </QueryClientProvider>
    </Router>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
