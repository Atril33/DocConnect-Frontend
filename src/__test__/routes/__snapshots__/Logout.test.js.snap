// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout component renders correctly 1`] = `
<div
  className="relative flex items-center justify-center h-screen bg-center bg-cover bg-home-background-mobile md:bg-home-background"
>
  <div
    className="w-full max-w-xs"
  >
    <h1
      className="mb-6 text-5xl font-bold text-center text-white"
    >
      Logout
    </h1>
    <form
      className="px-8 pt-6 pb-8 mb-4 bg-white rounded shadow-md"
      onSubmit={[Function]}
    >
      <div
        className="mb-4"
      >
        <p>
          Are you sure you want to log out?
        </p>
      </div>
      <div
        className="mb-4"
      >
        <button
          className="bg-green-400 hover:bg-green-700 text-white font-bold py-2 px-4 rounded w-full"
          type="submit"
        >
          Log out
        </button>
      </div>
      <p
        className="text-xs text-center text-gray-500"
      >
        OR
      </p>
      <p
        className="text-xs text-center text-gray-500"
      >
        Go back
         
        <a
          className="underline"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </p>
    </form>
  </div>
</div>
`;
